rust/src/lib.rs

```diff
@@ -107,12 +113,44 @@ unsafe fn wallet_db(
 /// - The memory referenced by `fsblock_db` must not be mutated for the duration of the function call.
 /// - The total size `fsblock_db_len` must be no larger than `isize::MAX`. See the safety
 ///   documentation of pointer::offset.
-fn block_db(fsblock_db: *const u8, fsblock_db_len: usize) -> Result<FsBlockDb, failure::Error> {
+fn block_db(fsblock_db: *const u8, fsblock_db_len: usize) -> anyhow::Result<FsBlockDb> {
     let cache_db = Path::new(OsStr::from_bytes(unsafe {
         slice::from_raw_parts(fsblock_db, fsblock_db_len)
     }));
     FsBlockDb::for_path(cache_db)
-        .map_err(|e| format_err!("Error opening block source database connection: {}", e))
+        .map_err(|e| anyhow!("Error opening block source database connection: {}", e))
+}
+
+/// Initializes global Rust state, such as the logging infrastructure and threadpools.
+///
+/// # Panics
+///
+/// This method panics if called more than once.
+#[no_mangle]
+pub extern "C" fn zcashlc_init_on_load() {
+    // Set up the tracing layer for the Apple OS logging framework.
+    let os_layer = os_log::layer("co.electriccoin.ios", "rust").with_filter(LevelFilter::INFO);
+
+    // Install the `tracing` subscriber.
+    tracing_subscriber::registry().with(os_layer).init();
+
+    // Log panics instead of writing them to stderr.
+    log_panics::init();
+
+    // Manually build the Rayon thread pool, so we can name the threads.
+    rayon::ThreadPoolBuilder::new()
+        .thread_name(|i| format!("zc-rayon-{}", i))
+        .build_global()
+        .expect("Only initialized once");
+
+    debug!("Rust backend has been initialized successfully");
+    cfg_if::cfg_if! {
```

Comment:
@str4d I think I'm not seeing any of these two messages being printed in the debugger console or in the OSLog console. They used to be .

Reply:
Oh, whoops: I'm setting the max log level to INFO above, so DEBUG level log messages would be completely ignored. I'll fix the PR to not limit log levels at all on the Rust side, and hopefully the Apple side limiting will work.